components/pkg-export-kubernetes/src/main.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate clap;
+extern crate env_logger;
+extern crate habitat_core as hcore;
+extern crate habitat_common as common;
+#[macro_use]
+extern crate log;
+
+use clap::App;
+use std::result;
+use std::str::FromStr;
+
+use hcore::PROGRAM_NAME;
+use hcore::env as henv;
+use hcore::package::PackageIdent;
+use common::ui::{Coloring, UI, NOCOLORING_ENVVAR, NONINTERACTIVE_ENVVAR};
+
+// Synced with the version of the Habitat operator.
+pub const VERSION: &'static str = "0.1.0";
+
+fn main() {
+    env_logger::init().unwrap();
+    let mut ui = ui();
+    if let Err(e) = start(&mut ui) {
+        let _ = ui.fatal(e);
+        std::process::exit(1)
+    }
+}
+
+fn ui() -> UI {
+    let isatty = if henv::var(NONINTERACTIVE_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Some(false)
+    } else {
+        None
+    };
+    let coloring = if henv::var(NOCOLORING_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Coloring::Never
+    } else {
+        Coloring::Auto
+    };
+    UI::default_with(coloring, isatty)
+}
+
+fn start(_ui: &mut UI) -> result::Result<(), String> {
+    let m = cli().get_matches();
+    debug!("clap cli args: {:?}", m);
+    let count = m.value_of("COUNT").unwrap_or("1");
+    // clap_app!() ensures that we do have the mandatory args so unwrap() is fine here
+    let pkg_ident_str = m.value_of("PKG_IDENT").unwrap();
+    let pkg_ident = match PackageIdent::from_str(pkg_ident_str) {
+        Ok(pi) => pi,
+        Err(e) => return Err(format!("{}", e)),
+    };
+    let image = m.value_of("IMAGE").unwrap_or(pkg_ident_str);
+
+    println!(
+        r###"---
```

Comment:
Might be good to use a separate template for this, e.g.

* https://github.com/habitat-sh/habitat/blob/64c6f5c2fb012bfc7c724eab3120d8f5240288a6/components/pkg-export-docker/defaults/Dockerfile.hbs
* https://github.com/habitat-sh/habitat/blob/64c6f5c2fb012bfc7c724eab3120d8f5240288a6/components/pkg-export-docker/src/docker.rs#L444

Also, it looks like you're sending this all to standard output; is that intentional?

Reply:
I think that printing the manifest to the standard output has its uses - you can redirect the output to the new file or to append it to an already existing one, since a single yaml file can contain many k8s manifests. Also, you can do something like `hab pkg export kubernetes foo/bar | kubectl apply -f -`.

Reply:
Perhaps we should have a '--output' parameter to control this. If the parameter is not set, we output to stdout.